airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -134,12 +138,32 @@ protected void acceptTracked(final AirbyteMessage message) throws Exception {
         return;
       }
 
-      // if the buffer flushes, update the states appropriately.
-      // TODO: ryankfu (if true, this is where bundling up logic to also commit to airbyte_raw table)
-      if (bufferingStrategy.addRecord(stream, message)) {
-        markStatesAsFlushedToTmpDestination();
+      final Optional flushType = bufferingStrategy.addRecord(stream, message);
+      // if present means that a flush occurred
+      if (flushType.isPresent()) {
+        if (flushType.equals(BufferFlushType.FLUSH_ALL)) {
```

Comment:
This should make SpotBugs happy:
```
final Optional<BufferFlushType> flushType = bufferingStrategy.addRecord(stream, message);
...
if (flushType.get().equals(BufferFlushType.FLUSH_ALL)) {
...
} else if (flushType.get().equals(BufferFlushType.FLUSH_SINGLE_STREAM)) {

```

Reply:
Actually I think `.get()` [throws an error](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get--) if the value is not present

My stylistic opinion here would be wrap the conditionals into a helper method `shouldMarkStateAsFlushedToDestination`

This would then become:

```
flushType.filter(shouldMarkStateAsFlushedToDestination).ifPresent(markStateAsFlushedToDestination);
```

Reply:
Also `shouldMarkStateAsFlushedToDestination` could look like this:

```
private boolean shouldMarkStateAsFlushedToDestination(BufferFlushType flushType) {
  switch(flushType) {
    case BufferFlushType.SINGLE_STREAM:
      return stateManager instanceOf DestStreamLifecycleManager;
    case BufferFlushType.FLUSH_ALL:
      return True;
    default:
      return False;
  }
}
```

Reply:
I'd suggest enabling this intellij inspection, since spotbugs is kind of annoying to work with 
![image](https://user-images.githubusercontent.com/5741425/212441921-e4c774a2-382b-4ce9-90f9-a3c08d339c30.png)

it's (correctly) pointing out that this if condition is never true, since you're comparing an enum to an optional